docs/adr/0004-framework.md

```diff
@@ -0,0 +1,68 @@
+# Use React Framework
```

Comment:
I would change the file name from "0004-framework" to something like "0004-frontend-framework" in order to diff with potential future ADRs about other types of framework decisions.

Reply:
@mansaj Great call-out. Adding this as a last step TODO as I don't wanna lose any of the current comments' context.

Reply:
@maxxcrawford Not sure if you still want to rename the ADR, post merge. 🤷

Reply:
Renamed!